contentcuration/contentcuration/frontend/shared/views/MarkdownEditor/plugins/registerCustomMarkdownField.js

```diff
@@ -0,0 +1,51 @@
+import Vue from 'vue';
+import vueCustomElement from 'vue-custom-element';
+import uuid from 'uuid/v4';
+
+export default VueComponent => {
+  const dashed = camel => camel.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase();
+  const name = dashed(VueComponent.name);
+  Vue.use(vueCustomElement);
+  Vue.customElement(name, VueComponent, {
+    // element is removed from document
+    disconnectedCallback() {
+      this.observer.disconnect();
+    },
+    // Vue instance is created
+    vueInstanceCreatedCallback() {
+      // by default, `contenteditable` will be false
+      this.setAttribute('contenteditable', !!VueComponent.contentEditable);
+
+      // a hack to prevent squire from merging custom element spans
+      // see here: https://github.com/nhn/tui.editor/blob/master/libs/squire/source/Node.js#L92-L101
+      this.classList.add(`markdown-field-${uuid()}`);
```

Comment:
Deep down in TUI's vendored fork of squire.js [lives a function called `areAlike`](https://github.com/nhn/tui.editor/blob/master/libs/squire/source/Node.js#L92-L101) which is used to merge `span` elements into a single span element... unless they have different `class` or `style` attributes.  

Adding a unique class to each custom field element stops TUI from doing this.  It's a hack, but a very critical one, since this was causing image/formula field elements to get globbed together, making one or the other effectively disappear.